crypto/evp/p_rsa.c

```diff
@@ -582,24 +613,24 @@ int EVP_PKEY_CTX_set_rsa_pss_keygen_mgf1_md(EVP_PKEY_CTX *ctx,
 }
 
 int EVP_PKEY_CTX_set_rsa_pss_saltlen(EVP_PKEY_CTX *ctx, int salt_len) {
-  return EVP_PKEY_CTX_ctrl(ctx, EVP_PKEY_RSA,
+  return EVP_RSA_PKEY_CTX_ctrl(ctx,
                            (EVP_PKEY_OP_SIGN | EVP_PKEY_OP_VERIFY),
                            EVP_PKEY_CTRL_RSA_PSS_SALTLEN, salt_len, NULL);
 }
 
 int EVP_PKEY_CTX_get_rsa_pss_saltlen(EVP_PKEY_CTX *ctx, int *out_salt_len) {
-  return EVP_PKEY_CTX_ctrl(ctx, EVP_PKEY_RSA,
+  return EVP_RSA_PKEY_CTX_ctrl(ctx,
                            (EVP_PKEY_OP_SIGN | EVP_PKEY_OP_VERIFY),
                            EVP_PKEY_CTRL_GET_RSA_PSS_SALTLEN, 0, out_salt_len);
 }
 
 int EVP_PKEY_CTX_set_rsa_keygen_bits(EVP_PKEY_CTX *ctx, int bits) {
-  return EVP_PKEY_CTX_ctrl(ctx, EVP_PKEY_RSA, EVP_PKEY_OP_KEYGEN,
+  return EVP_RSA_PKEY_CTX_ctrl(ctx, EVP_PKEY_OP_KEYGEN,
```

Comment:
Rant: This call actually allows requesting a RSA key with bit length as low as `256`. *sigh*
But our RSA implementation can generate keys down to that size, so it is probably a bigger puzzle figuring out if we can bump that.

Reply:
Is a SIM needed to track this?

Reply:
In short, medium term it is okay imo. Long-term. Well maybe it is a good idea to have it as a backlog item to about ponder about.